internal/prober/prober.go

```diff
@@ -145,22 +146,23 @@ func (p *Prober) checkAndTriggerScale(ctx context.Context, candidateNodeLeases [
 
 // shouldPerformScaleUp returns true if the ratio of expired node leases to valid node leases is less than
 // the NodeLeaseFailureFraction set in the prober config
-func (p *Prober) shouldPerformScaleUp(candidateNodeLeases []coordinationv1.Lease) bool {
+func (p *Prober) shouldPerformScaleUp(candidateNodeLeases []coordinationv1.Lease) (bool, *float64) {
 	if len(candidateNodeLeases) == 0 {
 		p.l.Info("No owned node leases are present in the cluster, performing scale up operation if required")
-		return true
+		return true, nil
 	}
 	var expiredNodeLeaseCount float64
 	for _, lease := range candidateNodeLeases {
 		if p.isLeaseExpired(lease) {
 			expiredNodeLeaseCount++
 		}
 	}
-	shouldScaleUp := expiredNodeLeaseCount/float64(len(candidateNodeLeases)) < *p.config.NodeLeaseFailureFraction
+	expiredFraction := expiredNodeLeaseCount / float64(len(candidateNodeLeases))
+	shouldScaleUp := expiredFraction < *p.config.NodeLeaseFailureFraction
 	if shouldScaleUp {
-		p.l.Info("Lease probe succeeded, performing scale up operation if required")
+		p.l.Info("Lease probe succeeded, performing scale up operation if required", "expiredFraction", expiredFraction, "threshold", *p.config.NodeLeaseFailureFraction)
```

Comment:
```suggestion
		p.l.Info("Lease probe succeeded, performing scale up operation if required", "expiredLeases", expiredFraction, "expirationThreshold", *p.config.NodeLeaseFailureFraction)
```
Also, I've put `*expirationFraction` in the value above as it was missing here.